### PR TITLE
feat: add `applicationCommandMention()` formatter

### PR DESCRIPTION
**Pull request by almeidx**
**Please describe the changes this PR makes and why it should be merged:**
- Added `applicationCommandMention()` formatted in /builders
- Added `ApplicationCommand#toString()`, which will use the function mentioned above to return the commands mention

Upstream:
- discord/discord-api-docs#5186

**Status and versioning classification:**
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: packages/discord.js/src/util/Formatters.js [hunk 1]
```diff
@@ -24,6 +25,16 @@ const {
...
 class Formatters extends null {
+  /**
+   * Formats an application command name and id into an application command mention.
+   * @method applicationCommandMention
+   * @memberof Formatters
+   * @param {string} commandName The name of the application command.
+   * @param {string} commandId The id of the application command.
+   * @returns {string}
+   */
+  static applicationCommandMention = applicationCommandMention;
```
Comment (ImRodry) [inline packages/discord.js/src/util/Formatters.js:36]: could you add this to ApplicationCommand#toString()?

Comment (almeidx) [reply packages/discord.js/src/util/Formatters.js:39]: And how would subcommands (and subcommand groups) be handled? Would they just be ignored?

Comment (ImRodry) [reply packages/discord.js/src/util/Formatters.js:39]: Well yeah there's no way to mention a subcommand

File: packages/builders/src/messages/formatters.ts [hunk 2]
```diff
@@ -182,6 +182,75 @@ export function roleMention<C extends Snowflake>(roleId: C): `<@&${C}>` {
...
+ * @param subcommandGroupName - The subcommand group name to format
+ * @param subcommandName - The subcommand name to format
+ * @param commandId - The application command ID to format
+ */
+export function applicationCommandMention<N extends string, G extends string, S extends string, I extends Snowflake>(
+	commandName: N,
+	subcommandGroupName: G,
+	subcommandName: S,
+	commandId: I,
+): `</${N} ${G} ${S}:${I}>`;
```
Comment (ImRodry) [inline packages/builders/src/messages/formatters.ts:198]: You shouldn't add this imo. It doesn't actually work, it just displays the text nicely but you can't click it.

Comment (almeidx) [reply packages/builders/src/messages/formatters.ts:198]: It works on my end

File: packages/discord.js/src/util/Formatters.js (hunk 1 above)
Comment (ImRodry) [reply packages/discord.js/src/util/Formatters.js:39]: like I said you can write any text so I think toString should only put the name of the command

Comment (almeidx) [reply packages/discord.js/src/util/Formatters.js:39]: 062bcccaf14d5c2a02693e3ece09dca1e8615af2

**Review by vladfrangu — APPROVED** [review]

**Comment by kyranet** [comment]
This PR needs a rebase.

**Comment by almeidx** [comment]
Rebased

**Review by iCrawl — APPROVED** [review]

**Comment by thehackerboi69github** [comment]
This should also be used in https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/structures/ChatInputCommandInteraction.js

**Comment by almeidx** [comment]
> This should also be used in https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/structures/ChatInputCommandInteraction.js

You cannot include options in command mentions

**Comment by thehackerboi69github** [comment]
> > This should also be used in https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/structures/ChatInputCommandInteraction.js
> 
> You cannot include options in command mentions
> 

That's true, but now that there are attachment options, `ChatInputCommandInteraction#toString` is not 100% working anymore, and now that there is an official application command mention format, I feel like we should use the official method instead of a custom, not fully working one.

**Review by kyranet — APPROVED** [review]

**Comment by almeidx** [comment]
Can't reopen this pull request now. Made a new one, #8546

**Pull request merged**